lib/tools/apk-utils.js

```diff
@@ -474,12 +474,76 @@ apkUtilsMethods.getDeviceLocale = async function () {
 
 /**
  * Set the locale name of the device under test.
+ * Compatibility for appium-android-driver 1.32.4-
  *
  * @param {string} locale - The name of the new device country.
  */
 apkUtilsMethods.setDeviceLocale = async function (locale) {
-  // this method is only used in API >= 23
-  await this.setDeviceSysLocale(locale);
+  let split_locale = locale.split("-");
+  await this.setDeviceLanguageCountry(split_locale[0], split_locale[1]);
+};
+
+/**
+ * Set the locale name of the device under test.
+ *
+ * @param {string} language - Language. format: [a-zA-Z]{2,8}. e.g. en, ja : https://developer.android.com/reference/java/util/Locale.html
+ * @param {string} country - Country. format: [a-zA-Z]{2} | [0-9]{3}. e.g. US, JP : https://developer.android.com/reference/java/util/Locale.html
```

Comment:
I've changed don't convert the language and the country to _upper case_ or _lower case_ in `setDeviceLanguageCountry` because they should accept `[a-zA-Z]` as Locale spec.